ui/pages/settings/flask/snaps-list-tab/snap-list-tab.component.js

```diff
@@ -32,14 +32,15 @@ const SnapListTab = ({
   dispatch,
 }) => {
   const t = useI18nContext();
-  if (viewingSnap) {
+  if (viewingSnap && currentSnap) {
     return (
       <ViewSnap
         snap={currentSnap}
         onToggle={(event) => onToggle(event, currentSnap)}
-        onRemove={() => {
-          onRemove();
-          dispatch(removeSnap(currentSnap.id));
+        onRemove={(event) => {
+          dispatch(removeSnap(currentSnap.id)).then(() => {
+            onRemove(event, currentSnap);
```

Comment:
Do we need the `event` and `currentSnap` arguments? They don't seem to be part of the function signature.